### PR TITLE
Basic Search Index support for h2

### PR DESCRIPTION
**Pull request by crisptrutski**
Very crude support, using LIKE instead of full text.

I evaluated both the built-in and Lucene extension full text support in h2, but neither produced reasonable ranking. The built-in version did not support completions, so it totally dead in the water for the command palette. Lucene requires adding new dependencies, but could probably improve our filtering.

For now this does no text ranking at all. One idea is to bring back in-memory ranking.

### CODE REVIEW

**Review by sansolovyov — APPROVED** [review]
Seems like a good first step :)

**Comment by crisptrutski** [comment]
* **#51049** <a href="https://app.graphite.dev/github/pr/metabase/metabase/51049?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#51042** <a href="https://app.graphite.dev/github/pr/metabase/metabase/51042?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/metabase/metabase/51042?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `master`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/metabase/search/appdb/index.clj [hunk 1]
```diff
@@ -51,23 +56,28 @@
...
-  (when table-name
-    (t2/exists? :information_schema.tables :table_name (name table-name))))
+(defn- table-name [kw]
+  (cond-> (name kw)
+    (= :h2 (mdb/db-type)) u/upper-case-en))
+
+(defn- exists? [table]
+  (when table
+    ;; TODO specialize the case based on the driver
+    (t2/exists? :information_schema.tables :table_name (table-name table))))
```
Comment (piranha) [inline src/metabase/search/appdb/index.clj:66]: This `TODO` seems to be implemented already?

**Review by sansolovyov — APPROVED** [review]
Still like it, even after all those changes 😁

**Pull request merged**